shared/api/cluster.go

```diff
@@ -0,0 +1,40 @@
+package api
+
+// Cluster represents high-level information about a LXD cluster.
+type Cluster struct {
+	Name string `json:"name" yaml:"name"`
+}
+
+// ClusterPut represents the fields required to bootstrap or join a LXD
+// cluster.
+//
+// API extension: cluster
+type ClusterPut struct {
+	Name          string        `json:"name" yaml:"name"`
+	Address       string        `json:"address" yaml:"address"`
+	Schema        int           `json:"schema" yaml:"schema"`
+	API           int           `json:"api" yaml:"api"`
+	TargetAddress string        `json:"target_address" yaml:"target_address"`
+	TargetCert    string        `json:"target_cert" yaml:"target_cert"`
+	TargetCA      []byte        `json:"target_ca" yaml:"target_ca"`
+	StoragePools  []StoragePool `json:"storage_pools" yaml:"storage_pools"`
+	Networks      []Network     `json:"networks" yaml:"networks"`
```

Comment:
This struct is confusing to me because I thought a whole bunch of those things were now moved to the internal call.
That is, the API client doesn't need to tell LXD about the storage pools or networks that it's got, nor its Schema version or API version. LXD itself already knows all of those things.

I'd have expected it to look like:

```go
type ClusterPut struct {
    ServerName string `json:"server_name" yaml:"server_name"`
    ClusterAddress string `json:"cluster_address" yaml:"cluster_address"`
    ClusterCertificate string `json:"cluster_certificate" yaml:"cluster_certificate"`
}
```

 - Address is core.https_address (PUT should fail if it's not set to something appropriate)
 - Schema is known internally
 - API is known internally
 - TargetCA isn't needed when we already pass the exact Certificate anyways
 - StoragePools is known internally
 - Networks is known internally

Reply:
Oh yeah, those fields were indeed leftovers, I should have moved them to the internal API. That's done now.

In addition to what you asked, I also renamed ClusterMember.Name to ClusterMember.ServerName, which seems more consistent. Those are individual commits, so feel free to strip them out if you wish.